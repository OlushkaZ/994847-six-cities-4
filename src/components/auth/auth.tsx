import * as React from 'react';
import {connect} from "react-redux";
import {Link} from 'react-router-dom';

interface Props {
  authorizationStatus: string;
  email: string;
}

const Auth: React.FC<Props> = ({authorizationStatus, email}) => {
  return (
    <nav className="header__nav">
      <ul className="header__nav-list">
        <li className="header__nav-item user">
          {(authorizationStatus === `NO_AUTH`)
            ? (
              <Link to={{pathname: `/sign-in`}} className="header__nav-link header__nav-link--profile">
                <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                <span className="header__login">Sign in</span>
              </Link>
            )
            : (
              <Link
                className="header__nav-link header__nav-link--profile"
                to={{pathname: `/favorites`}}
              >
                <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                <span className="header__user-name user__name">
                  {email}
                </span>
              </Link>
            )}
        </li>
      </ul>
    </nav>
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: state.user.authorizationStatus,
  email: state.user.email,
});

export {Auth};
export default connect(mapStateToProps)(Auth);

