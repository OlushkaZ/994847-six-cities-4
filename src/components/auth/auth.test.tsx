import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {Auth} from './auth';

it(`Should Auth render correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <Auth authorizationStatus={`AUTH`} email={`123@gmail.com`}/>
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should NoAuth render correctly`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <Auth authorizationStatus={`NO_AUTH`} email={``}/>
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
