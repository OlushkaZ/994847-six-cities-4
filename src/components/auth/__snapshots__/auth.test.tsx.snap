// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Auth render correctly 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="/favorites"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__user-name user__name"
        >
          123@gmail.com
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Should NoAuth render correctly 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="/sign-in"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__login"
        >
          Sign in
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
